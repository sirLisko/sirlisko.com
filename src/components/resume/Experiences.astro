---
import type { Resume } from "../../types";

interface Props {
  experiences: Resume["experiences"];
}

const { experiences } = Astro.props;
---

<section class="section experiences">
  <h2>Experiences</h2>
  {
    experiences.map((exp) => (
      <div>
        {exp.where && (
          <h3>
            {exp.where} {exp.when && <span>{exp.when}</span>}
          </h3>
        )}
        {exp.blurp.map((blr) => (
          <p>{blr}</p>
        ))}
      </div>
    ))
  }
</section>
