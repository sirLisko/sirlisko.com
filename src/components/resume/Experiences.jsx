import React from "react";
import PropTypes from "prop-types";

const Experiences = ({ experiences }) => (
  <article className="section experiences">
    <h1>Experiences</h1>
    {experiences.map(exp => (
      <div key={exp.where || exp.role}>
        {exp.where && (
          <h2>
            {exp.where} {exp.when && <span>{exp.when}</span>}
          </h2>
        )}
        {exp.blurp.map((blr, i) => (
          <p key={i}>{blr}</p>
        ))}
      </div>
    ))}
  </article>
);

Experiences.propTypes = {
  experiences: PropTypes.array.isRequired
};

export default Experiences;
