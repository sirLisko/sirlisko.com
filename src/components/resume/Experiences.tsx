import React from "react";

interface ExperiencesProps {
  experiences: Array<{
    where: string;
    blurp: string[];
    when?: string;
    role?: string;
  }>;
}

const Experiences = ({ experiences }: ExperiencesProps) => (
  <article className="section experiences">
    <h1>Experiences</h1>
    {experiences.map((exp) => (
      <div key={exp.where || exp.role}>
        {exp.where && (
          <h2>
            {exp.where} {exp.when && <span>{exp.when}</span>}
          </h2>
        )}
        {exp.blurp.map((blr, i) => (
          <p key={i}>{blr}</p>
        ))}
      </div>
    ))}
  </article>
);

export default Experiences;
