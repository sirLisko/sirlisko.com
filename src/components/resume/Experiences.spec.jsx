import React from "react";
import renderer from "react-test-renderer";

import Experiences from "./Experiences.jsx";

describe("Experiences Component", () => {
  let wrapper;
  const props = {
    experiences: [
      {
        role: "foo",
        blurp: ["..."],
      },
      {
        role: "foo",
        where: "bar",
        when: "May 2000 - July 2001",
        blurp: ["foo", "bar", "foobar"],
      },
    ],
  };

  beforeEach(() => {
    wrapper = renderer.create(<Experiences {...props} />).toJSON();
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
