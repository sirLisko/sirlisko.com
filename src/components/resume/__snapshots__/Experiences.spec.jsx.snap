// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiences Component should render properly 1`] = `
<article
  className="section experiences"
>
  <h1>
    Experiences
  </h1>
  <div>
    <h2>
      foo
       
    </h2>
    <p>
      ...
    </p>
  </div>
  <div>
    <h2>
      foo
       
      <span>
        @ 
        bar
      </span>
    </h2>
    <h4>
      May 2000 - July 2001
    </h4>
    <p>
      foo
    </p>
    <p>
      bar
    </p>
    <p>
      foobar
    </p>
  </div>
</article>
`;
